idtoken/idtoken.go

```diff
@@ -50,7 +51,9 @@ func NewClient(ctx context.Context, audience string, opts ...ClientOption) (*htt
 	if err != nil {
 		return nil, err
 	}
-	opts = append(opts, option.WithTokenSource(ts))
+	// skipping validation so added TokenSource will not conflict with user
+	// provided credentials.
+	opts = append(opts, option.WithTokenSource(ts), internaloption.SkipValidation())
```

Comment:
Bikeshed: I don't know what `SkipValidation` means here. Perhaps something more specific? `SkipDialSettingsValidation`? It's longer, but it's more clear what's happening.

Reply:
I think that is a fair call out. Being extra verbose here seems okay to me since this should only be used by us and promotes readability.